ui/src/app/utils/stores.tsx

```diff
@@ -30,6 +30,46 @@ interface ProfileStore {
 
 export const profileStore = atom<ProfileStore>({});
 
+export interface RuntimeOperation {
+  promise: Promise<any>;
+  operation: string;
+  aborter: AbortController;
+}
+
+export interface WorkspaceRuntimeOperationMap {
+  [workspaceNamespace: string]: RuntimeOperation;
+}
+
+export interface RuntimeOpsStore {
+  opsByWorkspaceNamespace: WorkspaceRuntimeOperationMap;
+}
+
+export const runtimeOpsStore = atom<RuntimeOpsStore>({opsByWorkspaceNamespace: {}});
+
+export const updateRuntimeOpsStoreForWorkspaceNamespace = (workspaceNamespace: string, runtimeOperation: RuntimeOperation) => {
+  const opsByWorkspaceNamespace = runtimeOpsStore.get().opsByWorkspaceNamespace;
+  opsByWorkspaceNamespace[workspaceNamespace] = runtimeOperation;
+  runtimeOpsStore.set({opsByWorkspaceNamespace: opsByWorkspaceNamespace});
```

Comment:
do we need to set the entire map for this to work? I would have thought that its sufficient to just update the map with the line above.

Reply:
Javascript is pass by value

Reply:
Is that still true for objects? I was able to modify an object with the follow code snippet. 

The behavior may be different in your case though because of atomic store.

```
function flipValue(map) {
  map['key'] = "newValue"
}

myMap = {'key': "oldValue"};
flipValue(myMap);

console.log("Key is: " + myMap['key'])
```
`Key is: newValue`

Reply:
Hm, ok. Perhaps it is atom related, then. It was definitely failing to update the store when I didn't explicitly set it.